extensions/Oversight/migrateToRevDel.php

```diff
@@ -0,0 +1,280 @@
+				$insertData[$fieldPrefix . 'user_text'] = isset( $userNames[$hiddenRow->hidden_user] ) ? $userNames[$hiddenRow->hidden_user] : $hiddenRow->hidden_user_text;
+				$insertData[$fieldPrefix . 'timestamp'] = $hiddenRow->hidden_timestamp;
+				$insertData[$fieldPrefix . 'minor_edit'] = $hiddenRow->hidden_minor_edit;
+				// Hide revision text, edit summary, editor's username/IP, even from admins.
+				$insertData[$fieldPrefix . 'deleted'] = Revision::DELETED_TEXT | Revision::DELETED_COMMENT | Revision::DELETED_USER | Revision::DELETED_RESTRICTED;
+
+				$rev = Revision::newFromRow( $hiddenRow );
+
+				$insertData[$fieldPrefix . 'len'] = $rev->getSize();
+				$insertData[$fieldPrefix . 'sha1'] = $rev->getSha1();
+
+				if ( $dbw->selectField(
+					'archive',
+					'ar_rev_id',
+					array(
+						'ar_rev_id' => $hiddenRow->hidden_rev_id,
+						'ar_sha1' => $insertData[$fieldPrefix . 'sha1'],
+						'ar_timestamp' => $hiddenRow->hidden_timestamp
+					),
+					__METHOD__
+				) ) {
+					$this->output( "Ignoring revision {$hiddenRow->hidden_rev_id} as it is already in the archive table.\n" );
+					$lastRevId = $hiddenRow->hidden_rev_id;
+					continue;
+				}
+
+				if ( $tableName == 'archive' ) {
+					$insertData['ar_parent_id'] = null;
+					$insertData['ar_namespace'] = $hiddenRow->hidden_namespace;
+					$insertData['ar_title'] = $hiddenRow->hidden_title;
+				} else {
+					$parentIdFromTables = array(
+						$dbw->selectRow(
+							'revision',
+							array( 'rev_id', 'timestamp' => 'rev_timestamp' ),
+							array(
+								'rev_page' => $hiddenRow->hidden_page,
+								'rev_timestamp < ' . $dbw->addQuotes( $hiddenRow->hidden_timestamp ),
+								'rev_id < ' . $dbw->addQuotes( $hiddenRow->hidden_rev_id )
+							),
+							__METHOD__,
+							array( 'ORDER BY' => 'rev_timestamp DESC' )
+						),
+						$dbw->selectRow(
+							'hidden',
+							array( 'rev_id' => 'hidden_rev_id', 'timestamp' => 'hidden_timestamp' ),
+							array(
+								'hidden_page' => $hiddenRow->hidden_page,
+								'hidden_timestamp < ' . $dbw->addQuotes( $hiddenRow->hidden_timestamp ),
+								'hidden_rev_id < ' . $dbw->addQuotes( $hiddenRow->hidden_rev_id )
+							),
+							__METHOD__,
+							array( 'ORDER BY' => 'hidden_timestamp DESC' )
+						),
+						$dbw->selectRow(
+							'archive',
+							array( 'rev_id' => 'ar_rev_id', 'timestamp' => 'ar_timestamp' ),
+							array(
+								'ar_page_id' => $hiddenRow->hidden_page,
+								'ar_timestamp < ' . $dbw->addQuotes( $hiddenRow->hidden_timestamp ),
+								'ar_rev_id < ' . $dbw->addQuotes( $hiddenRow->hidden_rev_id ),
+								'ar_namespace' => $hiddenRow->hidden_namespace,
+								'ar_title' => $hiddenRow->hidden_title
+							),
+							__METHOD__,
+							array( 'ORDER BY' => 'ar_timestamp DESC' )
+						)
+					);
+
+					$timestampsToRevIds = array();
+					foreach ( $parentIdFromTables as $parentIdFromTable ) {
+						if ( $parentIdFromTable != false ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "false" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)